pyanaconda/payload/dnf/payload.py

```diff
@@ -433,6 +434,39 @@ def _is_existing_repo_configuration(self, data):
         """
         return not data.url and data.name in self._dnf_manager.repositories
 
+    def _set_up_additional_repository(self, data):
+        """Set up sources for the additional repository."""
+        # Set up the NFS source with a substituted URL.
+        if data.url.startswith("nfs://"):
+            device_mount = self._get_mount_point(
+                constants.MOUNT_DIR,
+                data.name + "-nfs-device"
+            )
+            iso_mount = self._get_mount_point(
+                constants.MOUNT_DIR,
+                data.name + "-nfs-iso"
+            )
+            task = SetUpNFSSourceTask(
+                device_mount=device_mount,
+                iso_mount=iso_mount,
+                url=self._dnf_manager.substitute(data.url)
+            )
+            mount_point = task.run()
+            data.url = "file://" + mount_point
+
+    def _get_mount_point(self, *paths):
+        """Generate a mount point from specified paths."""
+        mount_point = join_paths(*paths)
+        self._mount_points.append(mount_point)
+        return mount_point
```

Comment:
That doesn't really generate the mount point, only its path, I think. So maybe call it `_get_mount_point_path`, and change the docstring for that?

Reply:
I have changed it, however, these methods are temporary.